rl4co/models/zoo/mdam/decoder.py

```diff
@@ -272,15 +274,15 @@ def _get_log_p(self, fixed, td, path_index, normalize=True):
         glimpse_v = fixed.glimpse_val + glimpse_val_dynamic
         logit_k = fixed.logit_key + logit_key_dynamic
 
-        # Compute the mask
+        # Compute the mask (NOTE: here it is the opposite due to custom attention implementation)
         mask = ~td["action_mask"]
```

Comment:
You're right, I think I skipped this due to lazyness but it can be further modularized. It seems no strange attention implementation was used in this model anyways!

Reply:
(As discussed, left as TODO for the future for the time being!)